app/package/controllers.py

```diff
@@ -330,70 +331,66 @@ def purge_data_package(publisher, package):
         return handle_error('GENERIC_ERROR', e.message, 500)
 
 
-@package_blueprint.route("/<publisher>/<package>/finalize",
-                         methods=["POST"])
+@package_blueprint.route("/upload", methods=["POST"])
 @requires_auth
-@is_allowed('Package::Create')
```

Comment:
I think this should still require authorization. Otherwise anyone can call this and reimport data. Amongst everything else it opens us to potential DDOS since the operation is expensive potentially.